thorlcr/activities/filter/thfilterslave.cpp

```diff
@@ -58,27 +58,40 @@ class CFilterSlaveActivity : public CFilterSlaveActivityBase, public CThorSteppa
 
     IHThorFilterArg *helper;
     unsigned matched;
+
 public:
     CFilterSlaveActivity(CGraphElementBase *container)
         : CFilterSlaveActivityBase(container), CThorSteppable(this)
     {
     }
-    void init(MemoryBuffer &data, MemoryBuffer &slaveData)
+    void init(MemoryBuffer &data, MemoryBuffer &slaveData) override
     {
         PARENT::init(data,slaveData);
         helper = static_cast <IHThorFilterArg *> (queryHelper());
     }
-    void start()
+    virtual void start() override
     {   
         ActivityTimer s(totalCycles, timeActivities);
         matched = 0;
-        abortSoon = !helper->canMatchAny();
-        PARENT::start();
+        if (helper->canMatchAny())
+            PARENT::start();
+        else
+        {
+            dataLinkStart();
+            abortSoon = true;
+            stop();
+        }
+    }
+    virtual void stop() override
+    {
+        if (!queryInputStopped(0))
```

Comment:
The equivalent roxie code will stop the input stream not the activity in filter.
Or do similar in the IF or CASE activities for inactive inputs during start()
[ Ultimately this often avoids the splitters needs to use a smart buffer. ]

Here (and in IF/CASE) in Thor, I'm stopping the input activity (which will stop the input stream).
The activity will not have been started, but you do want to chain through the input stop chain, so that upstream splitters know asap.
It probably could call inputStream->stop and not activity::stop instead, if I tracked whether act vs input had been stopped separately, but I don't for simplicity.

I'm not sure this causes any issues in practice?